server/src/main/java/org/opensearch/indices/ShardLimitValidator.java

```diff
@@ -85,17 +97,28 @@ public int getShardLimitPerNode() {
         return shardLimitPerNode.get();
     }
 
+    private void setIgnoreDotIndexes(boolean newValue) {
+        this.ignoreDotIndexes = newValue;
+    }
+
     /**
      * Checks whether an index can be created without going over the cluster shard limit.
+     * Validate shard limit only for non system indices as it is not hard limit anyways.
+     * Further also validates if the cluster.ignore_dot_indexes is set to true.
+     * If so then it does not validate any index which starts with '.' except data-stream index.
      *
      * @param indexName      the name of the index being created
      * @param settings       the settings of the index to be created
      * @param state          the current cluster state
      * @throws ValidationException if creating this index would put the cluster over the cluster shard limit
      */
     public void validateShardLimit(final String indexName, final Settings settings, final ClusterState state) {
-        // Validate shard limit only for non system indices as it is not hard limit anyways
-        if (systemIndices.validateSystemIndex(indexName)) {
+        /*
+        Validate shard limit only for non system indices as it is not hard limit anyways.
+        Further also validates if the cluster.ignore_dot_indexes is set to true.
+        If so then it does not validate any index which starts with '.'.
+        */
+        if ((systemIndices.validateSystemIndex(indexName) || validateDotIndex(indexName)) && !isDataStreamIndex(indexName)) {
```

Comment:
Isn't the first condition a subset of the dot index

Reply:
Yes it would be but a subset other criteria will still exists and thus that needs to be there.